tools/cldr-apps/src/main/java/org/unicode/cldr/web/STFactory.java

```diff
@@ -230,11 +231,41 @@ private VoteResolver<String> setValueFromResolver(String path, VoteResolver<Stri
             }
             delegate.removeValueAtDPath(path);
             if (value != null) {
+                value = reviseInheritanceAsNeeded(path, value);
```

Comment:
This is actually an odd place to update.  because this is the bottleneck for basically READING from the DB, not for STORING.  

I'm not sure this does what you want.  This won't affect the database. 

But again, I'm not sure of the goals.

Reply:
Given the update on the ticket. this is not in the right place.  This will cause the 'revised' value to vary over time instead of being stored as such in the DB.  It will affect the reading of the DB, but will affect it differently each time.

Also, this is a recursive problem.   Calling STFactory.make() indirectly calls `setValueFromResolver()`! 

As noted in the ticket, the `BallotBox.voteForValueWithType()` override on STFactory would be the right place to make this change (with the worker function pushed into a utility).

Reply:
@srl295 "this is the bottleneck for basically READING from the DB, not for STORING"

-- that's not clear to me. As noted in the ticket, the call stack when a user votes is:

putValueAtPath:272, XMLSource (org.unicode.cldr.util)
setValueFromResolver:233, STFactory$DataBackedSource (http://org.unicode.cldr.web/)
voteForValueWithType:1088, STFactory$PerLocaleData (http://org.unicode.cldr.web/)
voteForValueWithType:369, STFactory$PerLocaleData (http://org.unicode.cldr.web/)
handleVote:377, VoteAPIHelper (org.unicode.cldr.web.api)
vote:270, VoteAPI (org.unicode.cldr.web.api)

"set" and "put" both look like STORING, not READING -- so where does STORING really happen?

Reply:
"STFactory.make() indirectly calls setValueFromResolver()"

even if the CLDRFile is already cached?

Reply:
that cache is invalidated if the value changes.